packages/fxa-auth-server/lib/routes/linked-accounts.ts

```diff
@@ -139,6 +142,20 @@ export class LinkedAccountHandler {
       sessionToken: sessionToken.data,
     };
   }
+
+  async unlinkAccount(request: AuthRequest) {
+    if (!this.googleAuthClient) {
+      throw error.thirdPartyAccountError();
+    }
+    const uid = request.auth.credentials.uid;
+    if ((request.payload as any).provider.toLowerCase() === 'google') {
+      // TODO: here we'll also delete any session tokens created via a google login
+      await this.db.deleteLinkedGoogleAccount(uid);
+    }
+    return {
```

Comment:
If we don't recognize the provider, is it technically correct to return a success response?

Reply:
I think this might be ok because validate the providers coming in and would fail the request before it get here.